_posts/2021-11-12-github-copilot.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: Github Copilot - let AI write your code?
+author: arto
+excerpt: >
+  Using tools to increase effectiveness has been a long-standing trend in the software development community. Latest wave comes in form of GPT-3, OpenAI, and ultimately Github Copilot. Let's try to answer the question: Can AI write your code?
+tags:
+ - Software Development
+ - AI
+ - GPT-3
+ - OpenAI
+ - Github Copilot
+---
+
+_One day soon, AI will be writing the code. I, for one, welcome our new robot overlords._
+
+![Copilot helping with Markdown](/img/github-copilot/copilot_helping_with_markdown_3.png)
+*Copilot is writing this blog text for me*
+
+## What is Github Copilot?
+
+Right now it's an intellisense-kind of tool you can get in popular IDEs. Whereas traditional code suggestion plugins typically detect patterns in your code and suggest code snippets, Github Copilot is reading your code and trying to understand your context, comments, and method names, and then synthesizing code suggestions. Raw material for the suggestions comes from Github public repositories. But it's not just a copy-paste snippet.
+
+You might have seen GPT-3 based code generation demos before. Copilot is powered by OpenAI Codex, which is a new AI system, more capable than GPT-3. It's currently available as a technical preview, on request. It's expected that later on it will be released as a commercial offering, open for everyone.
+
+My interest in this tool comes from always wanting to improve efficiency and learning, so I applied for technical preview, and got accepted, and have now used this for a few weeks for various personal projects.
+
+## What can Github Copilot do for you?
+
+Well, quite a lot. Once you have installed the plugin and the Copilot has been enabled for you, you can start writing code, and it will be automatically filled in with code suggestions. You can ignore them and keep on coding as usual, or press the TAB key to accept the suggestions. You can also press Ctrl+Enter to get up to 10 different suggestions for the problem.
+
+Copilot is using the code you have written as a hint for the code suggestions. If you started writing a function, you'll get nice suggestions for the function, based on the name and parameters. If you started defining a variable, you get suggestions on how to populate it. You can also write comments, to explain in English what you want to accomplish.
+
+Let's see how Copilot could help you implement a simple temperature system conversion function. Easy mode is to start defining a function, let's do this in Python. Once you start writing the function name, you get the suggestion for implementation (in gray). You can accept it by pressing the TAB key.
```

Comment:
> Easy mode is to start defining a function, let's do this in Python.

"Using Python, let's start by defining a function." ?

Reply:
Need to formulate this slightly differently because rest of the text is defined as instructions for 'you' - but yes, can be simplified a bit, good find!